adsft/extraction.py

```diff
@@ -811,6 +811,11 @@ def extract_list(self, static_xpath, **kwargs):
                     normalise=True,
                     trim=True)
 
+                # to ignore facility tags missing xlink:href
+                # https://github.com/adsabs/ADSfulltext/issues/107
+                if text_content == 'None' or text_content == '':
+                    continue

```

Comment:
I worry that we are changing the general function `extract_list` which might be used for other purposes and maybe some other parts of the code is expecting these lists of Nones. Have you verified that? Alternatively, is there any other part that is specific to facilities where we could have this check?

Reply:
There isn't a place in the code that is specific for facilities besides the actual xpath, but I could put an if statement to only change this for facilities. The only other list is the dataset, and we don't do anything besides extract that in the fulltext pipeline. I'll discuss this change with whoever deals with the datasets (I believe this is Edwin) and give you an update.

Reply:
I investigated a bit too regarding the datasets, they are sent to master pipeline (https://github.com/adsabs/ADSPipelineMsg/blob/97970b07d707d5bf040d1a9fb543f41a1687142f/specs/fulltext_update.proto#L10) but then the closest that there is in the Solr schema is `data` (https://github.com/romanchyla/montysolr/blob/f7c2f6cbb1c0c20704b940abb68c2a4fb6e10988/contrib/examples/adsabs/server/solr/collection1/conf/schema.xml#L1366). I have the impression that `dataset` is never sent to anywhere else and it just stays in the master pipeline.

When expanding the `data` facet in the UI, it generates a query like:

```
curl 'https://ui.adsabs.harvard.edu/v1/search/query?__clearBigQuery=true&facet=true&facet.field=data_facet&facet.limit=20&facet.mincount=1&facet.offset=0&fl=id&q=star&rows=1&sort=date%20desc%2C%20bibcode%20desc' -H 'authorization: Bearer:<token>'
```

Which returns data that does not seem to come from the fulltext pipeline:

```
{"facet_counts": {"facet_heatmaps": {}, "facet_ranges": {}, "facet_fields": {"data_facet": ["SIMBAD", 162671, "NED", 35495, "MAST", 26932, "Vizier", 15837, "CDS", 14253, "ESA", 11883, "ESO", 10784, "HEASARC", 7785, "Chandra", 5639, "XMM", 3107, "IRSA", 3086, "Herschel", 2484, "GCPD", 2240, "Spitzer", 1928, "NOAO", 1866, "INES", 1737, "ALMA", 1443, "KOA", 1298, "ATNF", 1218, "NExScI", 1193]}, "facet_queries": {}, "facet_intervals": {}}, "responseHeader": {"status": 0, "QTime": 27, "params": {"sort": "date desc, bibcode desc", "facet": "true", "facet.mincount": "1", "rows": "1", "facet.field": "data_facet", "facet.limit": "20", "__clearBigQuery": "true", "q": "star", "start": "0", "wt": "json", "facet.offset": "0", "fl": "id"}}, "response": {"start": 0, "numFound": 527032, "docs": [{"id": "18678641"}]}}
```
So, unless you discover something different on your end, it looks like we would be good to go with your current changes. I'll change the status to approved, so that if you don't find any objection, you can go ahead and merge it.

Reply:
Thanks for looking into this, I'll merge as what I learned yesterday aligns with this conclusion.